Islandora-CLAW/controlled_access_terms
# Update EDTF field for 2018 spec compliance

**GitHub Issue**: /[Islandora-CLAW/CLAW/issues/995](https://github.com/Islandora-CLAW/CLAW/issues/995)

# What does this Pull Request do?

Fixes the Widget and Formatter for the EDTF Field Type to match [the 2018 spec](https://www.loc.gov/standards/datetime/edtf.html).

# What's new?
* Adds an EDTFUtils class to consolidate EDTF-related code.
* Updates the validation code (now in EDTFUtils) to meet the 2018 spec.
* Updates the Widget and Formatter to use EDTFUtils and meet the spec.
* Adds an update hook to bring existing content in EDTF fields to meet the 2018 spec.

# How should this be tested?
* Take a site with controlled_access_terms enabled and add some content to EDTF fields (repository items, taxonomy terms, whatever). They all comply with the 2012 spec.
* Apply the PR
* Run updates (either 'siteurl/update.php' or `drush updb -y`)
* Clear cache.
* Check your content. All the values should display correctly (even if exact wording and display has been adjusted) and viewing them in the edit window should show new signifiers. E.g. '?~' is now '%' and unspecified values 'u' are now 'X'. 

# Interested parties
@Islandora-CLAW/committers
